confidential/confidential.go

```diff
@@ -298,3 +306,25 @@ func SurjectionProof(input SurjectionProofInput) ([]byte, error) {
 
 	return secp256k1.SurjectionProofSerialize(ctx, proof)
 }
+
+//SatoshiToElementsValue method converts Satoshi value to Elements value
+func SatoshiToElementsValue(val uint64) (result [9]byte, err error) {
+	unconfPrefix := byte(1)
+	b := bytes.NewBuffer([]byte{})
+	if err = common.BinarySerializer.PutUint64(b, binary.LittleEndian, val); err != nil {
+		return
+	}
+	copy(result[:], append([]byte{unconfPrefix}, bufferutil.ReverseBytes(b.Bytes())...))
+
+	return
+}
+
+//ElementsToSatoshiValue method converts Elements value to Satoshi value
+func ElementsToSatoshiValue(val [9]byte) (result uint64, err error) {
+	reverseValueBuffer := [confidentialValue - 1]byte{}
```

Comment:
Here the first thing to do is to ensure that `val` hash the correct `0x01` prefix, otherwise return a `invalid prefix` error
```suggestion
      if val[0] != byte(1) {
           err = errors.New("invalid prefix")
           return
      }
	reverseValueBuffer := [confidentialValue - 1]byte{}
```

Reply:
Done